kolibri/plugins/learn/assets/src/views/exam-page/index.vue

```diff
@@ -116,13 +119,24 @@
             answer: answer.answerState,
             correct: answer.correct,
           });
-          this.setAndSaveCurrentExamAttemptLog(this.content.id, this.itemId, attempt);
+          return this.setAndSaveCurrentExamAttemptLog(this.content.id, this.itemId, attempt);
         }
+        return Promise.resolve();
+      },
+      goToQuestion(questionNumber) {
+        this.saveAnswer();
         this.$router.push({
           name: PageNames.EXAM,
           params: { channel_id: this.channelId, id: this.exam.id, questionNumber },
         });
       },
+      submitExam() {
+        if (!this.submitModalOpen) {
+          this.saveAnswer().then(() => {
+            this.toggleModal();
```

Comment:
Actually, could be further reduced to `this.saveAnswer().then(this.toggleModal);`